src/redux/reducers/currentProjectReducer.test.ts

```diff
@@ -39,14 +40,35 @@ describe("Current Project Reducer", () => {
         expect(result).toBeNull();
     });
 
-    it("Load Project Assets merges assets into current asset set", () => {
+    it("Updating connection not in use by current project performs noop", () => {
+        const currentProject = MockFactory.createTestProject("1");
+        const state: IProject = currentProject;
+        const unrelatedConnection = MockFactory.createTestConnection("Unrelated Connection");
+        const action = saveConnectionAction(unrelatedConnection);
+        const result = reducer(state, action);
+        expect(result).toEqual(currentProject);
+    });
+
+    it("Updating connection used by current project is updated in curren project", () => {
+        const currentProject = MockFactory.createTestProject("1");
+        const state: IProject = currentProject;
+
+        const sourceConnection = {...currentProject.sourceConnection};
+        sourceConnection.description += "updated";
+
+        const action = saveConnectionAction(sourceConnection);
+        const result = reducer(state, action);
+
+        expect(result.sourceConnection).toEqual(action.payload);
```

Comment:
should we also test that current project has change?

Reply:
Updated expectations.